kinto/core/authorization.py

```diff
@@ -151,22 +155,32 @@ def __init__(self, request):
         is_on_resource = (
             service is not None and hasattr(service, "viewset") and hasattr(service, "resource")
         )
+        self._resource = None
         if is_on_resource:
             self.resource_name = request.current_resource_name
             self.on_plural_endpoint = getattr(service, "type", None) == "plural"
 
-            # Try to fetch the target object. Its existence will affect permissions checking.
-            if not self.on_plural_endpoint and request.method.lower() in (
+            # Check if this request targets an individual object.
+            # Its existence will affect permissions checking (cf `_find_required_permission()`).
+            # There are cases where the permission is not directly related to the HTTP method,
+            # For example:
+            # - with POST on plural endpoint, with an id supplied
+            # - with PUT on an object, which can either be creation or update
+            is_write_on_object = not self.on_plural_endpoint and request.method.lower() in (
                 "put",
                 "delete",
                 "patch",
-            ):
-                resource = service.resource(request=request, context=self)
-                try:
-                    # Save a reference, to avoid refetching from storage in resource.
-                    self.current_object = resource.model.get_object(resource.object_id)
-                except storage_exceptions.ObjectNotFoundError:
-                    pass
+            )
+            is_post_on_plural = self.on_plural_endpoint and request.method.lower() == "post"
+            if is_write_on_object or is_post_on_plural:
+                # We instantiate the resource to determine the object targetted by the request.
```

Comment:
```suggestion
                # We instantiate the resource to determine the object targeted by the request.
```

Sorry... I don't make the rules :slightly_smiling_face: